### PR TITLE
The Great Workening

### PR DESCRIPTION
**Pull request by dsmith328**
## About The Pull Request
Makes Abnormality Works based off your Attribute level, NOT your total level. Yes, this was how it worked. It was a legacy solution to a thing that was yet to be implemented.

## Why It's Good For The Game
It really should just be this way.

## Changelog
:cl:
add: Added a check for attribute level to decide what work level you perform.
/:cl:

### CODE REVIEW

**Review by Kitsunemitsu — APPROVED** [review]
Okay, looks perfect.

**Comment by dsmith328** [comment]
Shows Attribute Levels now

File: code/datums/abnormality/datum/abnormality.dm [hunk 1]
```diff
@@ -163,7 +163,17 @@
...
+		var/work_level = 1
+		switch(workType)
+			if(ABNORMALITY_WORK_INSTINCT)
+				work_level = clamp(round(get_attribute_level(user, FORTITUDE_ATTRIBUTE)/20), 1, 5)
+			if(ABNORMALITY_WORK_INSIGHT)
+				work_level = clamp(round(get_attribute_level(user, PRUDENCE_ATTRIBUTE)/20), 1, 5)
+			if(ABNORMALITY_WORK_ATTACHMENT)
+				work_level = clamp(round(get_attribute_level(user, TEMPERANCE_ATTRIBUTE)/20), 1, 5)
+			if(ABNORMALITY_WORK_REPRESSION)
+				work_level = clamp(round(get_attribute_level(user, JUSTICE_ATTRIBUTE)/20), 1, 5)
```
Comment (EgorDinamit) [inline code/datums/abnormality/datum/abnormality.dm:175]: All of it should be either generalized or turned into a proc of its own.

File: code/datums/abnormality/datum/abnormality.dm [hunk 2]
```diff
@@ -163,7 +163,17 @@
 		return 0
 	var/acquired_chance = available_work[workType]
 	if(islist(acquired_chance))
-		acquired_chance = acquired_chance[get_user_level(user)]
```
Comment (EgorDinamit) [inline code/datums/abnormality/datum/abnormality.dm:166]: For example, if you don't want to make it a proc of its own:
`var/work_level = clamp(round(get_attribute_level(user, work_type)/20), 1, 5)`
`acquired_chance = acquired_chance[work_level]`

Comment (EgorDinamit) [reply code/datums/abnormality/datum/abnormality.dm:166]: Wait, I'm dumb. There does, however, exist a define to turn work type names into attribute names. Use it.

**Review by EgorDinamit — APPROVED** [review]

**Pull request merged**